lib/run_part.c

```diff
@@ -61,7 +61,8 @@ int run_parts (const char *directory, const char *name, const char *action)
 		struct stat  sb;
 
 		if (asprintf(&s, "%s/%s", directory, namelist[n]->d_name) == -1) {
-			fprintf(stderr, "could not allocate memory\n");
+			fprintf(shadow_logfd, "%s: could not allocate memory\n",
+			    namelist[n]->d_name);
```

Comment:
But there are also perror(3) calls.  The commit should also justify why those still use stderr.

Reply:
Or I just replace them as well. ;) Done.

Reply:
I'd like at least a justification of why not unify the other way around.  :)

Reply:
https://github.com/shadow-maint/shadow/blob/master/lib/shadowlog_internal.h#L5

Reply:
I'd accept arguing about keeping perror after the fork, but simply writing into stderr from a fork isn't clean either. Otherwise the rational is that the logging infrastructure is there, both cases were available in that file and the majority in lib/ uses the logging infrastructure (as expected).

Reply:
I don't say it's wrong, but the commit message should say something.  Something like that would work.

Reply:
Adjusted